api/include/opentelemetry/trace/provider.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <atomic>
+
+#include "opentelemetry/nostd/unique_ptr.h"
+#include "opentelemetry/trace/noop.h"
+#include "opentelemetry/trace/tracer_provider.h"
+
+namespace opentelemetry
+{
+namespace trace
+{
+
+class DefaultTracerProvider final : public opentelemetry::trace::TracerProvider
+{
+public:
+  opentelemetry::trace::NoopTracer *const GetTracer(
+      nostd::string_view library_name,
+      nostd::string_view library_version  = "",
+      const TracerOptions &tracer_options = {}) override
+  {
+    return tracer_.get();
+  }
+
+private:
+  DefaultTracerProvider()
+      : tracer_{nostd::unique_ptr<opentelemetry::trace::NoopTracer>(
+            new opentelemetry::trace::NoopTracer)}
+  {}
+  nostd::unique_ptr<opentelemetry::trace::NoopTracer> tracer_;
```

Comment:
This should be a shared_ptr.

Reply:
Thanks for the feedback! Still getting my head around when it is safe to use STL stuff. Seems like anything that touches the public interface shouldn't be STL. As a sanity check, this is fine because it's only being used in `GetTracer` to fetch the underlying pointer, right?

Reply:
I think it's fine to have private data types using STL.

I posted this [comment](https://github.com/open-telemetry/opentelemetry-cpp/issues/4#issuecomment-584374975) to the C++ requirements issue to try to clarify where you need ABI stability.